packages/discord.js-utilities/src/lib/PaginatedMessage.ts

```diff
@@ -272,7 +283,9 @@ export class PaginatedMessage {
 	 * @param user The user that reacted to the message.
 	 */
 	protected async handleCollect(author: User, channel: TextChannel | NewsChannel, reaction: MessageReaction, user: User): Promise<void> {
-		await reaction.users.remove(user);
+		if ((channel as TextChannel).permissionsFor(channel.client.user?.id || '')?.has('MANAGE_MESSAGES')) {
+			await reaction.users.remove(user);
+		}
```

Comment:
```suggestion
		if (isGuildBasedChannel(channel) && channel.client.user && channel.permissionsFor(channel.client.user.id).has('MANAGE_MESSAGES')) {
			await reaction.users.remove(user);
		}
```

Don't typecast, that's extremely unsafe and this will crash in DMs because `permissionsFor` is not a method on `DMChannel`